substrate/frame/support/src/traits/tokens/fungible/item_of.rs

```diff
@@ -380,36 +380,64 @@ impl<
 		value: Self::Balance,
 		precision: Precision,
 	) -> Result<Debt<AccountId, Self>, DispatchError> {
-		<F as fungibles::Balanced<AccountId>>::deposit(A::get(), who, value, precision)
-			.map(|debt| Imbalance::new(debt.peek()))
+		<F as fungibles::Balanced<AccountId>>::deposit(A::get(), who, value, precision).map(
+			|inner_debt| {
+				let debt = Imbalance::new(inner_debt.peek());
+				fungibles::Imbalance::forget(inner_debt);
+				debt
+			},
```

Comment:
Would be nice to put this into a conversion function, so that it can become `<F as fungibles::Balanced<AccountId>>::deposit(A::get(), who, value, precision).map(conversion_function)`.